### ISSUE TITLE
Question Regarding Dataset Curation

### ISSUE THREAD

**Issue by zl3466** [tulerfeng/Video-R1#9]
Hello, thank you very much for open sourcing this great work!

I have two questions regarding your training dataset:

1. How did you determine the `reward` value in your training json dataset, especially for OCR and free-form questions? Your paper explained that **each question's reward value is calculated bewtween model answer and GT during training**, which I suppose is for supervised tuning. Are the `reward` in json dataset used for GRPO?

2. How did you generate the ground truth for "process" message?

Thanks in advance for your help.

**Comment by tulerfeng** [comment]
The reward is used for GRPO rather than SFT. During reinforcement learning (RL) training, we require the model to generate outputs in the format \<think\>`process`\<think\>\<answer\>`solution`\<answer\>. The reward is computed based on the distance between the generated `solution` and the ground truth. This encourages the model to learn how to generate a more effective `process` that ultimately leads to a better `solution` and higher reward.

For the COT annotation process that generates the process component, please refer to ./src/generate_cot_vllm.py

**Comment by zl3466** [comment]
> Thanks for your question.
> 
> The reward is used for GRPO rather than SFT. During reinforcement learning (RL) training, we require the model to generate outputs in the format <think>`process`<think><answer>`solution`<answer>. The reward is computed based on the distance between the generated `solution` and the ground truth. This encourages the model to learn how to generate a more effective `process` that ultimately leads to a better `solution` and higher reward.
> 
> For the COT annotation process that generates the process component, please refer to ./src/generate_cot_vllm.py

Thank you so much for the quick reply!

Do you mean the `reward` value in the **dataset json file** is calculated between the GT and the base model before training? 

I understand how you can calculate the reward between GT and model output, but since you created the dataset before training out a model via GRPO, what model output did you use to calculate that reward in dataset json?

**Comment by tulerfeng** [comment]
Oh no, sorry about that — we forgot to remove the reward values from the dataset JSON file. These rewards were generated during the COT annotation phase to help filter out low-quality process entries. We'll make sure to remove them.

**Comment by zl3466** [comment]
Thank you very much again for the response! To follow up on my earlier question regarding throught process ground truth, I noticed that in your `./src/generate_cot_vllm.py`, you used Qwen2.5VL 72B to generate the gt "process" message (the thought process between <think> tags). By doing this, are you assuming that the response generated by 72B is always acceptable Ground Truth for thought process? 
There's also a line in the `./src/generate_cot_vllm.py` where you did
```
            if think_chain:
                sample["process"] = f"<think>{think_chain}</think>"
```
and no else condition. Does this mean there's a chance an example's "process" value can be empty?

**Comment by tulerfeng** [comment]
> By doing this, are you assuming that the response generated by 72B is always acceptable Ground Truth for thought process?

Yes, if the answer produced by the 72B model is correct, we consider its reasoning process as the ground truth for SFT training.

> If so, is the loss a cross entropy loss between the "process" and model output, with the "process" content treated as a next-token-predction task?

Yes, during the SFT cold start, we have the model directly mimic the `process + answer` generated by the 72B model. In the subsequent RL training, however, the model is encouraged to explore and discover a better `process` on its own.

> Does this mean there's a chance an example's "process" value can be empty?

Yes, in certain cases, the 72B model fails to adhere closely to the prompt instructions, which prevents us from extracting a usable process from its output.

**Comment by zl3466** [comment]
> Yes, in certain cases, the 72B model fails to adhere closely to the prompt instructions, which prevents us from extracting a usable process from its output.

Thanks again for the prompt reply!
Did you exclude such failure cases from your dataset? As it appears that all examples in the dataset have the "process" entry properly filled.

**Comment by tulerfeng** [comment]
Yes, we exclude failure cases in our released dataset.
